analyzers/src/SonarAnalyzer.Common/Rules/RemoveObsoleteCodeBase.cs

```diff
@@ -0,0 +1,42 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarAnalyzer.Rules;
+
+public abstract class RemoveObsoleteCodeBase<TSyntaxKind> : SonarDiagnosticAnalyzer<TSyntaxKind>
+    where TSyntaxKind : struct
+{
+    private const string DiagnosticId = "S1133";
+
+    protected override string MessageFormat => "Do not forget to remove this deprecated code someday.";
+
+    protected RemoveObsoleteCodeBase() : base(DiagnosticId) { }
```

Comment:
There is a big overlap to ObsoleteAttributesNeedExplanation:

https://github.com/SonarSource/sonar-dotnet/blob/c1175674ce58abb27dd4c07a1b334311e51540f8/analyzers/src/SonarAnalyzer.Common/Rules/ObsoleteAttributesNeedExplanationBase.cs#L30-L44

What should we do @andrei-epure-sonarsource 
* Merge this PR and do follow up to merge the two rules
* Merge the two rules as part of this PR

Reply:
if you mean merging the implementation of the rules, I'd suggest to do it in a separate PR

if you mean to merge the rules themselves, it should be a discussion at LT level as these are already implemented in other languages, too

Reply:
I meant merging the implementations. I created https://github.com/SonarSource/sonar-dotnet/issues/6698 to track the work.